tools/webidl_binder.py

```diff
@@ -626,10 +628,13 @@ def make_call_args(i):
     basic_return = 'return ' if constructor or return_type != 'Void' else ''
     return_prefix = basic_return
     return_postfix = ''
+    storage_attribute = ''
     if non_pointer:
       return_prefix += '&'
+    if thread_safe:
+      storage_attribute = 'thread_local '
```

Comment:
Maybe we should just always emit this, when TLS gets lowered away when not building with thread support?

Reply:
We could, and you're right that if you build without `-pthread` the resulting code is the same:

https://godbolt.org/z/6cE6T8a47

When adding `-pthread` there is an extra instruction which adds a bit of overhead:

![image](https://github.com/user-attachments/assets/1208ee1c-ba7f-4366-b36f-17bbeba5ae58)

I cannot predict if people are ok with this overhead. Their library may use multiple threads internally, but maybe they don't interact with the library from multiple threads. In that case there's no benefit in adding the `thread_local` attribute.

Reply:
I think if you are doing a pthread build we should just default to being thread safe.

Reply:
Ok, in that case the change becomes really simple. I've pushed the new code.